### PR TITLE
Moving more data functions to datafactory

### PR DESCRIPTION
**Pull request by sthirugn**
Test results:
```sh
# nosetests tests/foreman/cli/test_environment.py

----------------------------------------------------------------------
Ran 14 tests in 152.948s

OK (skipped=2)


# nosetests tests/foreman/cli/test_roles.py

----------------------------------------------------------------------
Ran 4 tests in 1.393s

OK (skipped=4)

# nosetests tests/foreman/cli/test_user.py

----------------------------------------------------------------------
Ran 91 tests in 509.287s

OK (skipped=60)

# nosetests tests/foreman/ui/test_environment.py

----------------------------------------------------------------------
Ran 5 tests in 166.457s

OK

# nosetests tests/robottelo/test_datafactory.py
....
----------------------------------------------------------------------
Ran 4 tests in 0.024s

OK

```

### CODE REVIEW

File: tests/foreman/cli/test_roles.py [hunk 1]
```diff
@@ -33,7 +23,7 @@ def test_positive_create_role_1(self):
         @BZ: 1138553
 
         """
-        for name in valid_names():
+        for name in generate_strings_list(length=10):
```
Comment (sthirugn) [inline tests/foreman/cli/test_roles.py]: I changed this to make this consistent with the UI tests.  Also now html tests will be included.

**Comment by sthirugn** [comment]
Rebased

File: robottelo/datafactory.py [hunk 2]
```diff
@@ -161,3 +203,19 @@ def valid_names_list():
         u' {0}'.format(gen_string('utf8')),
         u'{0} '.format(gen_string('utf8')),
     ]
+
+
+@datacheck
+def valid_users_list():
```
Comment (elyezer) [inline robottelo/datafactory.py]: s/valid_users_list/valid_usernames_list?

Comment (elyezer) [inline robottelo/datafactory.py]: Also for further improvement, we may use valid_usernames, valid_ids, etc... If, for definition, data factories return list of data points than there is not need to add list to the name.

**Comment by elyezer** [comment]
ACK, my comments are not blocker.

File: robottelo/datafactory.py (hunk 2 above)
Comment (sthirugn) [inline robottelo/datafactory.py]: +1 Good idea for future improvement

**Comment by sthirugn** [comment]
Comments updated and rebased.

**Comment by elyezer** [comment]
ACK

File: robottelo/datafactory.py [hunk 3]
```diff
@@ -133,6 +150,31 @@ def valid_data_list():
...
 @datacheck
+def valid_emails_list():
+    """Returns a list of valid emails."""
+    return [
+        u'{0}@example.com'.format(gen_string('alpha')),
+        u'{0}@example.com'.format(gen_string('alphanumeric')),
+        u'{0}@example.com'.format(gen_string('numeric')),
+        u'{0}@example.com'.format(gen_string('alphanumeric', 48)),
+        u'{0}+{1}@example.com'.format(gen_alphanumeric(), gen_alphanumeric()),
+        u'{0}.{1}@example.com'.format(gen_alphanumeric(), gen_alphanumeric()),
```
Comment (JacobCallahan) [inline robottelo/datafactory.py]: why the difference between gen_string and gen_alphanumeric?

Comment (sthirugn) [inline robottelo/datafactory.py]: Nothing but just ease of use of using gen_alphanumeric.  But I agree, it should be consistently used.

Comment (JacobCallahan) [inline robottelo/datafactory.py]: either way is fine with me. one is more consistent one is more compact.

Comment (sthirugn) [inline robottelo/datafactory.py]: I removed the usage of gen_alphanumeric() and used gen_string instead.  Also this is the only place where gen_alphanumeric() is used so I have removed that from import too :)

**Comment by sthirugn** [comment]
@JacobCallahan comment addressed and rebased.

**Comment by JacobCallahan** [comment]
ACK

**Pull request merged**